bot/cogs/clean.py

```diff
@@ -138,7 +152,14 @@ def predicate_regex(message: Message) -> bool:
         # Now let's delete the actual messages with purge.
         self.mod_log.ignore(Event.message_delete, *message_ids)
         for channel in channels:
-            messages += await channel.purge(limit=amount, check=predicate)
+            if until_message:
+                for i in range(0, len(messages), 100):
+                    # while purge automatically handles the amount of messages
+                    # delete_messages only allows for up to 100 messages at once
+                    # thus we need to paginate the amount to always be <= 100
+                    await channel.delete_messages(messages[i:i + 100])
+            else:
+                messages += await channel.purge(limit=amount, check=predicate)
```

Comment:
Since you've essentially written all the code needed, can we get rid of `purge`? It redundantly iterates message history. We already have message objects from our own history search, so let's use those. That being said, one strange thing `purge` does is use a single delete strategy if messages are older than 2 weeks. I don't know if that's important.

Reply:
Reason why it does single deletes is because you can't delete messages older than 14 days via bulk delete endpoint, I didn't include that edge case in the code as (at least to me) it appears unlikely for moderators to attempt and purge messages older than that.
That being said attempts of deleting the messages older than 14 days via `delete_messages` will raise following error:
```python
discord.errors.HTTPException: 400 Bad Request (error code: 50034): You can only bulk delete messages that are under 14 days old.
```

Reply:
Do you think we can do away with `purge` though?

Reply:
I don't think `purge` is necessary considering we iterate the `history` either way,  I am all up to rewriting current system so it doesn't use `purge` in separate PR as it isn't exactly the point of this PR